torch/distributed/_spmd/gm_transformation.py

```diff
@@ -33,12 +32,12 @@ def __call__(self, gm: fx.GraphModule) -> Callable:
             )
             enable_graph_optimization_dump(graph_folder)
 
-        iter_gm = IterGraphModule(gm)
+        iter_gm = IterGraphModule(gm, enable_inductor=self.enable_inductor)
         if self.enable_graph_optimization:
             comm_fusion_with_concat(iter_gm, 100)
             schedule_comm_wait(iter_gm)
             remove_copy_from_optimizer(iter_gm)
-        iter_gm.freeze_cross_iter_movement()
+        # Must be called after we are not going to move the graphs
         iter_gm.setup(self.num_iters)
```

Comment:
future idea not related to this PR directly - seems like 'finalize' would be clearer name than setup here (setup implies first step, vs this is last step before locking down graphs and also this means having multiple setup references - could be confused with the 'setup' graph as an entity vs this as a function).

Reply:
Thanks for the suggestion. Right now we are unable to change the naming as it is already used by some internal code. Will change this accordingly afterward!